packages/ember-data/lib/system/relationships/state/has_many.js

```diff
@@ -16,6 +16,25 @@ ManyRelationship.prototype = Ember.create(Relationship.prototype);
 ManyRelationship.prototype.constructor = ManyRelationship;
 ManyRelationship.prototype._super$constructor = Relationship;
 
+ManyRelationship.prototype.setup = function(data) {
+  var key   = this.key,
+      value = data[key];
+
+  if (data.links && data.links[key]) {
+    this.updateLink(data.links[key]);
+  }
+
+  if (this.buildURL) {
+    var link = this.store.buildURLForHasMany(this.key, this.inverseKey, this.record);
```

Comment:
If the URL is dynamic, we should probably not calculate it upfront, because it might depend on data that changes in the meantime.